modules/calib3d/src/fundam.cpp

```diff
@@ -1039,4 +1039,24 @@ void cv::convertPointsHomogeneous( InputArray _src, OutputArray _dst )
         convertPointsToHomogeneous(_src, _dst);
 }
 
+double cv::sampsonDistance(InputArray _pt1, InputArray _pt2, InputArray _F) {
+    CV_Assert(_pt1.type() == CV_64F && _pt1.type() == CV_64F && _F.type() == CV_64F);
+    CV_DbgAssert(_pt1.rows() == 3 && _F.size() == Size(3, 3) && _pt1.rows() == _pt2.rows());
+
+    const Vec3d& pt1(_pt1.getMat().at<Vec3d>(0));
```

Comment:
thanks! Actually, in the case when _pt1/_pt2/_F are not physical Mat's nor arrays compatible with it (e.g. they are matrix expressions), the code may produce invalid result or even crash. There should be separate declaration of Mat's before you use .at&lt;&gt; etc.

<pre>
Mat pt1_m = _pt1.getMat();
const Vec3d pt1 = pt1_m.at<Vec3d>(0);
...
</pre>
this way you may sure that the temporary matrix is destroyed after pt1 and therefore the pointer to data stays valid during processing